dist/scripts/controllers/HomepageCtrl.js

```diff
@@ -1,18 +1,48 @@
 (function() {
-    function HomepageCtrl(Room, Message) {
-        
+    function HomepageCtrl(Room, Message, BlocChatCookies) {
+
         this.rooms = Room;
-        
+
         this.addRoom = function(newRoomName) {
             Room.addRoom(newRoomName); //need to figure out how to clear newRoomName
         };
-        
+
         this.messages = Message;
-        
+
         this.activeRoomMessages = '';
+
+        this.activeRoomId = '';
+
+        this.cookies = BlocChatCookies;
+
+        this.setUsername = function(newUsername) {
+            BlocChatCookies.setUsername(newUsername);
+        };
+
+        this.currentUser = this.cookies.currentUser;
+        this.resetUsernameCookies = this.cookies.resetCookies;
```

Comment:
Same with lines 22 and 23, we can access this directly through `this.cookies` rather than creating additional variables.